File: vs/src/CodeStream.VisualStudio/Models/Host.Protocol.Vsls.cs
namespace CodeStream.VisualStudio.Models
{
    public class LiveShareInviteToSessionRequest
    {
        public string UserId { get; set; }
        public bool CreateNewStream { get; set; }
    }

    public class LiveShareInviteToSessionRequestType : RequestType<LiveShareInviteToSessionRequest>
    {
        public const string MethodName = "host/vsls/invite";
        public override string Method => MethodName;
    }

    public class LiveShareJoinSessionRequest
    {
        public string Url { get; set; }
    }

    public class LiveShareJoinSessionRequestType : RequestType<LiveShareJoinSessionRequest>
    {
        public const string MethodName = "host/vsls/join";
        public override string Method => MethodName;
    }

    public class LiveShareStartSessionRequest
    {
        public string StreamId { get; set; }
        public string ThreadId { get; set; }
        public bool CreateNewStream { get; set; }
    }

    public class LiveShareStartSessionRequestType : RequestType<LiveShareStartSessionRequest>
    {
        public const string MethodName = "host/vsls/start";
        public override string Method => MethodName;
    }
}

File: shared/ui/ipc/host.protocol.vsls.ts
import { RequestType } from "vscode-jsonrpc";
import { IpcRoutes } from "./webview.protocol.common";

export interface LiveShareInviteToSessionRequest {
	userId: string;
	createNewStream: boolean;
}

export interface LiveShareInviteToSessionResponse {}

export const LiveShareInviteToSessionRequestType = new RequestType<
	LiveShareInviteToSessionRequest,
	LiveShareInviteToSessionResponse,
	void,
	void
>(`${IpcRoutes.Host}/vsls/invite`);

export interface LiveShareJoinSessionRequest {
	url: string;
}

export interface LiveShareJoinSessionResponse {}

export const LiveShareJoinSessionRequestType = new RequestType<
	LiveShareJoinSessionRequest,
	LiveShareJoinSessionResponse,
	void,
	void
>(`${IpcRoutes.Host}/vsls/join`);

export interface LiveShareStartSessionRequest {
	streamId: string;
	threadId: string;
	createNewStream: boolean;
}

export interface LiveShareStartSessionResponse {}

export const LiveShareStartSessionRequestType = new RequestType<
	LiveShareStartSessionRequest,
	LiveShareStartSessionResponse,
	void,
	void
>(`${IpcRoutes.Host}/vsls/start`);
